tag_reader.go

```diff
@@ -71,12 +74,37 @@ func (r *tagReader) ReadAll() ([]*Tag, error) {
 		})
 	}
 
-	r.sortTags(tags)
-	r.assignPreviousAndNextTag(tags)
+	switch r.sortBy {
+	case "date":
+		r.sortTags(tags)
+		r.assignPreviousAndNextTag(tags)
+	case "semver":
+		filterSemVerTags(&tags)
```

Comment:
See comment on line 90

```suggestion
		r.filterSemVerTags(&tags)
```